src/components/ha-web-rtc-player.ts

```diff
@@ -101,15 +118,50 @@ class HaWebRtcPlayer extends LitElement {
 
     console.timeLog("WebRTC", "end clientConfig", clientConfig);
 
-    const peerConnection = new RTCPeerConnection(clientConfig.configuration);
+    this._peerConnection = new RTCPeerConnection(clientConfig.configuration);
 
     if (clientConfig.dataChannel) {
       // Some cameras (such as nest) require a data channel to establish a stream
       // however, not used by any integrations.
-      peerConnection.createDataChannel(clientConfig.dataChannel);
+      this._peerConnection.createDataChannel(clientConfig.dataChannel);
+    }
+
+    this._peerConnection.onnegotiationneeded = () =>
+      this._startNegotiation(clientConfig);
+
+    this._peerConnection.onicecandidate = this._handleIceCandidate;
+    this._peerConnection.oniceconnectionstatechange =
+      this._iceConnectionStateChanged;
```

Comment:
Should we use `addEventListener` ? 

We are also mixing lambdas with methods that are locally assigned as lambdas (ie `startNegotiation`). Let's be consistent :)

Reply:
What would be the benefit of using `addEventListener`? We never want more than 1 listener per event anyway here.

Reply:
it feels so ugly to not use it 😬